### PR TITLE
1524: added structured data for acceptCharset

### PR DESCRIPTION
**Pull request by tobia80**
Removed AcceptCharset

### CODE REVIEW

**Comment by devsprint** [comment]
@thehonesttech  Mozilla is recommending to not support this header: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Charset

File: zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala [hunk 1]
```diff
@@ -0,0 +1,72 @@
+package zio.http.model.headers.values
+
+sealed trait AcceptCharset {
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala]: Let's move this to a new object `object HeaderValues` inside `zio.http.model`.

File: zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+   * Unicode character set
+   */
+  object `US-ASCII` extends AcceptCharset {
+    override val raw: String = "US-ASCII"
+  }
+
+  /**
+   * ISO Latin Alphabet No. 1, a.k.a. ISO-LATIN-1
+   */
+  object `ISO-8859-1` extends AcceptCharset {
```
Comment (jdegoes) [inline zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala]: I'm not sure about this convention. I see it in some Scala projects. But others would do `ISO_8859_1`, which I think I have a slight preference for.

Comment (adamgfraser) [reply zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala]: We use `_` in ZIO Core.

**Comment by tobia80** [comment]
> @thehonesttech Mozilla is recommending to not support this header: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Charset

Should I remove every accept-charset reference everywhere then?

**Comment by devsprint** [comment]
> > @thehonesttech Mozilla is recommending to not support this header: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Charset
> 
> Should I remove every accept-charset reference everywhere then?

Either we removed it totally or we might define a single value: UTF_8 and to ignore the raw value received.

File: zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala (hunk 1 above)
Comment (devsprint) [reply zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala]: @jdegoes the `HeaderValues` object will become very large. Keeping the `headers.values` package will allow easier parallelisation of development of the header values, i.e. less conflicts during merge ...

**Review by jdegoes — APPROVED** [review]

File: zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala (hunk 1 above)
Comment (jdegoes) [reply zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala]: How about this:

```scala
sealed trait HeaderValue
object HeaderValue {
  final case class Age(...) extends HeaderValue
  ...
}
```

This is a nice and logical way to organize them even if they share no similarities for now. 

It also let's us say at some point, `final case class Multiple(headerValues: Chunk[HeaderValue])` which we can't say right now.

File: zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala (hunk 2 above)
Comment (jdegoes) [reply zio-http/src/main/scala/zio/http/model/headers/values/AcceptCharset.scala]: All right, let's replicate that here.

**Review by jdegoes — APPROVED** [review]

**Pull request merged**